src/sysoptions.h

```diff
@@ -3,6 +3,13 @@
  * This file is only included from options.h
  *******************************************************************/
 
+/* TODO: find a better workaround */
+#ifdef __CYGWIN__
+#ifndef HAVE_SYS_RANDOM_H
+#define HAVE_SYS_RANDOM_H 1
```

Comment:
If you still have config.log handy, can you send it? It might show why sys/random.h isn't being found.